internal/scheduler/jobdb/reconciliation.go

```diff
@@ -0,0 +1,248 @@
+package jobdb
```

Comment:
This file contains the reconciliation logic previously in `syncState`. The long-term goal is to move all reconciliation logic into one place and then return armadaevents instead of `JobStateTransitions` as we do now.